File: app/src/main/java/com/plweegie/android/squash/data/RepoRepository.java
package com.plweegie.android.squash.data;

import android.arch.lifecycle.LiveData;

import com.plweegie.android.squash.utils.AppExecutors;

import java.util.List;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class RepoRepository {
    private final RepoDao mRepoDao;
    private final AppExecutors mExecutors;

    @Inject
    public RepoRepository(RepoDatabase database, AppExecutors executors) {
        mRepoDao = database.repoDao();
        mExecutors = executors;
    }

    public LiveData<List<RepoEntry>> getAllFavorites() {
        return mRepoDao.getFavorites();
    }

    public List<RepoEntry> getAllFavoritesDirectly() {
        return mRepoDao.getFavoritesDirectly();
    }

    public void addFavorite(RepoEntry repo) {
        mExecutors.diskIO().execute(() -> {
            mRepoDao.insertFavorite(repo);
        });
    }

    public void deleteRepo(long repoId) {
        mExecutors.diskIO().execute(() -> {
            mRepoDao.deleteSelected(repoId);
        });
    }

    public void deleteAllRepos() {
        mExecutors.diskIO().execute(() -> {
            mRepoDao.deleteAll();
        });
    }
}

File: app/src/main/java/com/plweegie/android/squash/data/RepoRepository.kt
package com.plweegie.android.squash.data

import androidx.lifecycle.LiveData
import com.plweegie.android.squash.rest.GitHubService
import java.util.concurrent.ExecutorService
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class RepoRepository @Inject constructor(database: RepoDatabase,
                                         private val network: GitHubService) {

    private val repoDao: RepoDao = database.repoDao()

    val allFavorites: LiveData<List<RepoEntry>>
        get() = repoDao.favorites

    suspend fun fetchRepos(userName: String, page: Int) =
            network.getRepos(userName, page)

    suspend fun getFavoritesAsync() = repoDao.getFavoritesAsync()

    suspend fun addFavorite(repo: RepoEntry) {
       repoDao.insertFavorite(repo)
    }

    suspend fun deleteRepo(repoId: Long) {
       repoDao.deleteSelected(repoId)
    }

    suspend fun deleteAllRepos() {
        repoDao.deleteAll()
    }
}
